packages/block-editor/src/components/inserter/block-patterns-tab/pattern-category-previews.js

```diff
@@ -70,27 +70,27 @@ export function PatternCategoryPreviews( {
 				if ( category.name === allPatternsCategory.name ) {
 					return true;
 				}
+
 				if (
 					category.name === myPatternsCategory.name &&
 					pattern.type === PATTERN_TYPES.user
 				) {
 					return true;
 				}
+
 				if ( category.name !== 'uncategorized' ) {
 					return pattern.categories?.includes( category.name );
 				}
 
 				// The uncategorized category should show all the patterns without any category
 				// or with no available category.
-				const availablePatternCategories =
-					pattern.categories?.filter( ( cat ) =>
-						availableCategories.find(
-							( availableCategory ) =>
-								availableCategory.name === cat
-						)
-					) ?? [];
+				if ( pattern.categories ) {
```

Comment:
This might relate to the comment from @glendaviesnz.

Should it be `if ( pattern.categories?.length ) {`?

Having said that, when there are any `pattern.categories` this will return early, so the `pattern.categories.some` code below would never be called when the array has values. Maybe it was supposed to be `if ( ! pattern.categories?.length ) {`.

It wasn't something modified in this PR, but thought I'd mention ... I think the code could be more readable if the `if` statement on line 81 is flipped to `if ( category.name === 'uncategorized' ) {`. I'd be happy to follow up with that change if it doesn't happen in this PR.

Reply:
The problem is that the condition needs flipping, thanks for noticing this 🙂 It should be `if ( ! pattern.categorires ) return true`. The idea is to provide a shortcut for the case where `pattern.categories` field is not there. In the previous code the `categories?.filter` expression evaluated to `undefined`, then it was defaulted to `[]`, and then the `.length === 0` comparison of this array leads to `true` return value. A four step process shortened to one step.

I also implemented the suggestion for `if ( name = 'uncategorized' )`. All the special cases are now evaluated inside `if` blocks, and the general case is handled last.